### ISSUE TITLE
UI Rendering issue

### ISSUE THREAD

**Issue by github59173** [shirokumacode/overwatch-omnic-rewards#24]
Not sure if this has to do with the OS I'm running but it has been a problem since 1.3. The UI elements seem to not get deleted when drawing new stats and eventually gets locked up because of this. Running qubes os fedora 35 with python 3.10. See picture for more details. 
![brokeuirender](https://user-images.githubusercontent.com/79297055/172070659-3fc29a8d-4f2a-4086-8a9a-e0ee6b00108c.png)

**Comment by Zensonaton** [comment]
Can confirm this on Manjaro Linux with KDE DE.

**Comment by shirokumacode** [comment]
Sorry for the late response, I only started looking into this late yesterday given there was no OWL league last week. 

This bug gave me a severe headache. I am assuming that both of you are using the executable, right?

So, building the app:
- locally on my pc (Arch) using system PyQt --> bug not present
- locally on my pc (Arch) using PyQt from pip --> bug not present
- using Github Actions (Ubuntu) using system PyQt --> bug present (this is the release since v1.2.1)
- using Github Actions (Ubuntu) using PyQt from pip --> bug not present (this was the release before v1.2.1)

This is most likely a Qt/Ubuntu package bug (or Pyinstaller, but seems unlikely). Possibly present since v1.2.1 or v1.3 as you recall. Will probably dive more into this when I have time and open a bug report on Ubuntu/Qt side.

Now I could just build the app with PyQt from pip as I was doing before v1.2.1 but that comes with a drawback: PyQt pip version has a bug that doesn't grab the desktop session properly. What this means is that if you shutdown/suspend/log off while the app is "watching", the app won't quit gracefully and thus won't write the watch time to file.
Or I could rewrite the part where I refresh the values and be hopeful that the bug goes away. That part of the code is poorly written but as it is, should have worked because I am literally removing the whole text widget, creating a new one and adding it. This is obviously a bad approach for multiple reasons. 

I have made a patch where I rewrote the stats value refresher function and it seemed to solve the issue. I will test it more and if all seems ok I'll release a new version where this is fixed.
